lib/class-convertercontroller.php

```diff
@@ -210,23 +268,68 @@ public function conversion_initialize() {
 	 *
 	 * @return array Conversion batch data.
 	 */
-	public function conversion_get_batch_data() {
+	public function get_conversion_batch_data() {
+		$has_incomplete_conversions = ! $this->conversion_processor->is_queued_conversion() && $this->conversion_processor->has_incomplete_conversions();
+
 		return rest_ensure_response(
 			[
-				'ids'       => $this->conversion_processor->set_next_conversion_batch_to_queue(),
-				'thisBatch' => max( $this->conversion_processor->get_conversion_queued_batches() ),
-				'maxBatch'  => $this->conversion_processor->get_conversion_max_batch(),
+				'ids'                      => $this->conversion_processor->set_next_conversion_batch_to_queue(),
+				'thisBatch'                => max( $this->conversion_processor->get_conversion_queued_batches() ),
+				'maxBatch'                 => $this->conversion_processor->get_conversion_max_batch(),
+				'hasIncompleteConversions' => $has_incomplete_conversions,
```

Comment:
After running the conversion for the first time, the `Warning: certain posts were not converted successfully.` message does not appear. On subsequent conversions (before repairing the data, so it's still inconvertible), the message appears. 

I've tried debugging it, but didn't get past this point here. My hunch is that `$has_incomplete_conversions` value here is the one that is erroneously `false`. Is that possible?

Reply:
Ah, awesome catch! I did a lot of code reshuffling and lost track of the sequence of execution which sets the state. Updated now, should work fine.